guides/source/getting_started.md

```diff
@@ -189,1952 +109,2183 @@ of the files and folders that Rails creates by default:
+When you visit a product's show action, you'll see the new caching lines in your Rails server logs:
 
-In fact, this is very close to the syntax that Rails uses to declare this
-association. You've already seen the line of code inside the `Comment` model
-(app/models/comment.rb) that makes each comment belong to an Article:
+```bash
+Read fragment views/products/show:a5a585f985894cd27c8b3d49bb81de3a/products/1-20240918154439539125 (1.6ms)
+Write fragment views/products/show:a5a585f985894cd27c8b3d49bb81de3a/products/1-20240918154439539125 (4.0ms)
+```
 
-```ruby
-class Comment < ApplicationRecord
-  belongs_to :article
-end
+The first time we open this page, Rails will generate a cache key and ask the cache store if it exists. This is the `Read fragment` line.
+
+Since this is the first page view, the cache does not exist so the HTML is generated and written to the cache. We can see this as the `Write fragment` line in the logs.
+
+Refresh the page and you'll see the logs no longer contain the `Write fragment`.
+
+```bash
+Read fragment views/products/show:a5a585f985894cd27c8b3d49bb81de3a/products/1-20240918154439539125 (1.3ms)
 ```
 
-You'll need to edit `app/models/article.rb` to add the other side of the
-association:
+The cache entry was written by the last request, so Rails finds the cache entry on the second request. Rails also changes the cache key when records are updated to ensure that it never renders stale cache data.
 
-```ruby
-class Article < ApplicationRecord
-  has_many :comments
+Learn more in the [Caching with Rails](caching_with_rails.html) guide.
 
-  validates :title, presence: true
-  validates :body, presence: true, length: { minimum: 10 }
-end
+Adding CSS & JavaScript
+-----------------------
+
+CSS & JavaScript are core parts of building web applications, so let's learn how to use them with Rails.
+
+### Propshaft
+
+Rails' asset pipeline is called Propshaft. It takes your CSS, JavaScript, images, and other assets and serves them to your browser. In production, Propshaft digests your assets so they can be cached indefinitely for performance.
+
+Let's modify `app/assets/stylesheets/application.css` and change our font to sans-serif.
+
+```css
+body {
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+nav {
+  display: flex;
+  gap: 0.5rem;
+}
+
+img {
+  max-width: 800px;
+}
+
+.notice {
+  color: green;
+}
 ```
 
-These two declarations enable a good bit of automatic behavior. For example, if
-you have an instance variable `@article` containing an article, you can retrieve
-all the comments belonging to that article as an array using
-`@article.comments`.
+Refresh your page and you'll see the CSS has been applied.
 
-TIP: For more information on Active Record associations, see the [Active Record
-Associations](association_basics.html) guide.
+### Importmaps
 
-### Adding a Route for Comments
+Rails uses Importmaps for JavaScript by default. This allows you to write modern JavaScript modules with no build steps.
```

Comment:
I wondered if this might be a bit advanced for beginners to grasp. Could go with something like: 

```suggestion
Rails uses Importmaps for JavaScript by default. This allows you to write modern JavaScript modules directly in your Rails application.
```

Reply:
Hey, "import map" is written that way in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap (two words, lower case).

For example, "An import map is a JSON object that allows developers...".